services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,59 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  if (thermalMgrQueueHandle == NULL) {
+    return ERR_CODE_INVALID_STATE;
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) == errQUEUE_FULL) {
+    return ERR_CODE_QUEUE_FULL;
+  }
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event = {.type = THERMAL_MGR_EVENT_INTERRUPT_TRIGGERED};
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t event;
+    float temperature;
+    error_code_t err;
+
+    if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) != pdTRUE) {
+      continue;
+    }
+
+    if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD) {
+      err = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temperature);
+      if (err == ERR_CODE_SUCCESS) {
+        addTemperatureTelemetry(temperature);
+        printConsole("%d\n", temperature);
+      } else {
+        printConsole("%d", err);
```

Comment:
We use the macro `LOG_ERROR_CODE(err)` to log the error code. Update this and the one below